test/main/utilities.js

```diff
@@ -0,0 +1,43 @@
+QUnit.module( "QUnit.objectType" );
+
+function buildFunctions() {
+	var objects = [ function( ) { } ];
+
+	try {
+		var arrow = new Function( "return () => {};" )( );
+		objects.push( arrow );
+	} catch ( e ) { }
+
+	try {
+		var async = new Function( "return async function( ) { };" )( );
+		objects.push( async );
+	} catch ( e ) { }
```

Comment:
@leobalter - HA! Yes, I started by trying to use `eval`, but I do not think that it allows returning the eval'ed value. This is important for us, because we have to ensure that having the syntax does not throw during evaluation of the whole file (and therefore miss all these tests).

Have I missed something?  Is there a better way?